packages/providers/src/transaction-request/transaction-request.ts

```diff
@@ -297,34 +297,36 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
    * @param predicate - Predicate bytes.
    * @param predicateData - Predicate data bytes.
    */
-  addMessageInput(message: MessageCoin, predicate?: AbstractPredicate) {
+  private addMessageInput(message: MessageCoinResource) {
     const { recipient, sender, amount } = message;
 
     const assetId = BaseAssetId;
 
-    let witnessIndex;
+    const input: MessageTransactionRequestInput = {
+      ...message,
+      type: InputType.Message,
+      sender: sender.toB256(),
+      recipient: recipient.toB256(),
+      amount,
+      witnessIndex: 0,
+      predicate: undefined,
+      predicateData: undefined,
```

Comment:
I reckon we can remove this explicit undefined set

Reply:
Done